### ISSUE TITLE
slower whith  multiple threads 

### ISSUE THREAD

**Issue by huangshanongfushan** [SYSTRAN/faster-whisper#599]
Hi,
I am using a GeForce 4060 to transcript a 80 seconds audio and it cost 10 seconds to get the result with 1 thread.
But When I use 3 Python threads to call transcript (one model and set num_workers  4) , It cost 40 seconds to get all results, 
why it becomes slower , it seems it did not worked in parallel.

Thanks a lot

**Comment by trungkienbkhn** [comment]
Yes it's normal behavior. Because in fact, python cannot run threads in parallel, this limitation is actually enforced by GIL. The Python Global Interpreter Lock (GIL) prevents threads within the same process to be executed at the same time. Multithread will run as time sharing as below:
![Screenshot from 2023-12-01 09-33-18](https://github.com/SYSTRAN/faster-whisper/assets/48075757/dcf0b8da-405a-4f09-9963-f1977a448032)
In the [ctranslate2](https://github.com/OpenNMT/CTranslate2/#key-features) project, some techniques to improve infer speed are also mentioned, which you can refer to.
Hope that's helpful to you.

**Comment by trungkienbkhn** [comment]
Yes. You can set num_workers higher if have multi gpu to improve the global throughput when using multiple thread.
For more information, you can refer to the [Ctranslate2 docs](https://opennmt.net/CTranslate2/parallel.html#data-parallelism) (num_workers is equivalent to the inter_thread variable)

**Comment by joiemoie** [comment]
> Yes. You can set num_workers higher if have multi gpu to improve the global throughput when using multi threads. For more information, you can refer to the [Ctranslate2 docs](https://opennmt.net/CTranslate2/parallel.html#data-parallelism) (num_workers is equivalent to the inter_threads variable)

I was wondering if you had a sense for what my bottlenecks could be. I tried my 4090, one in an AMD EPYC 7302 (16 cores, 3.3 GHz) with DDR4 ram and one in my i9 11700 KF (8 cores, getting up to 4.5 ish GHz) with DDR5 ram. It runs faster in my I9 setup, but I can’t tell if that’s because higher clock speed.

In general, I see that the GR and SM utilization are lower on the EPYC 7302 setup, which is likely causing the slowdown. But I’m not sure why there is lower utilization.
